flow/project.py

```diff
@@ -1287,10 +1286,20 @@ def hi_all(*jobs):
             _parent_class = parent_class
 
             def __call__(self, func):
+                # Have to traverse the mro to ensure that func is already an operation and that
+                # self.condition is not.
                 operation_functions = [
-                    operation[1]
-                    for operation in self._parent_class._collect_operations()
+                    func for name, func in self._parent_class._collect_operations()
                 ]
+                if func not in operation_functions:
+                    _deprecated_warning(
```

Comment:
This will raise instead of warn in the future, right? In other words, we still have to do this check in the future.

Reply:
Yes or something like it.